osu.Game/Overlays/Toolbar/Toolbar.cs

```diff
@@ -220,15 +224,30 @@ public ToolbarBackground()
                 };
             }
 
-            protected override bool OnHover(HoverEvent e)
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                inputManager = GetContainingInputManager();
+                showGradient.BindValueChanged(_ => updateState(), true);
+            }
+
+            protected override void Update()
             {
-                gradientBackground.FadeIn(transition_time, Easing.OutQuint);
-                return true;
+                base.Update();
+
+                var currentMousePosition = inputManager.CurrentState.Mouse.Position;
+                // ensure that the gradient is not shown if the mouse is above the window.
+                // this is relevant when the background is moving due to the toolbar being toggled.
+                showGradient.Value = currentMousePosition.Y >= 0 && Contains(inputManager.CurrentState.Mouse.Position);
```

Comment:
Not really liking this solution.

I'm not sure if I can come up with something better, but I would make a transparent container (above the buttons) that passes through hover input and make that toggle the gradient.

Reply:
You mean in front of the buttons? I guess that has a nonzero chance of working, although I'm not actually sure. I'll try tomorrow.

I guess I could also mess with the positional input queue to make the background receive positional inputs with priority...

Reply:
Positional input queue cannot be messed with outside of framework. So [0d70f2c](https://github.com/ppy/osu/pull/21937/commits/0d70f2c0fd3d070b49c6ce4225a88bed63932392) is about as nice as I can make this look, I think. It's still pretty dreadful, though.